Sources/Basics/ByteString+Extensions.swift

```diff
@@ -0,0 +1,27 @@
+/*
+ This source file is part of the Swift.org open source project
+ Copyright (c) 2020 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import TSCBasic
+
+extension ByteString {
+    /// A lowercase, hexadecimal representation of the SHA256 hash
+    /// generated for the byte string's contents.
+    ///
+    /// This property uses the CryptoKit implementation of
+    /// Secure Hashing Algorithm 2 (SHA-2) hashing with a 256-bit digest, when available,
+    /// falling back on a native implementation in Swift provided by TSCBasic.
+    public var sha256Checksum: String {
```

Comment:
Would it make sense to return the actual data here, and then the call site can convert it to hex, or base-whatever, etc?  The representation seems orthogonal to the computation.

Reply:
Looking at the existing call sites, we only call `SHA256().hash(bytes)` to produce a hexadecimal representation. Returning `ByteString` here would be more flexible, but that may be unnecessary. Really, it's a matter of taste; I'm happy with any solution that makes it easy to opt-in to a faster hash function when available.

Reply:
It's fine to leave this in this PR.  I just have a general preference for separating out functionality into orthogonal parts, but since this is already an extension ByteString, it's fine to leave as is.

ByteString came about back before String had a UTF-8 backing, so performance was a problem in going between UTF-8 data and String.  My sense is that we could probably transition to String here over time, at which point we might want to revisit this.

Reply:
A big 👍 for transitioning away from `ByteString` at some point. When I started working on the codebase for registry support, I found the use of TSC / SPM-specific APIs instead of Swift Standard Library and Foundation (and now [System](https://github.com/apple/swift-system)) counterparts to be a frequent point of frustration. Many of these APIs were necessary at the time (e.g. `JSON` before Codable),  but have now become technical debt.